### ISSUE TITLE
`prefer-node-protocol` should not flag non-node imports

### ISSUE THREAD

**Issue by jason-ha** [sindresorhus/eslint-plugin-unicorn#2293]
When a replacement node feature package is imported, as may be the case to support node features in non-node environments, those imports should not be flagged.

Rule: `prefer-node-protocol`

# Issue repro

package.json
```json
	"dependencies": {
		"@types/events": "^3.0.0",
		"events": "^3.1.0"
	}
```

```typescript
import { EventEmitter } from "events";
```

## Questionable

- If tsconfig compilerOptions types includes "node".
- Other dependencies depend on node.

In these cases, tsc will use the node types rather than the types for that package. It probably makes sense for these to be raised as issues, that deserve suppression, node: correction, or other consideration.

# Context

In a project, an import had a line suppression for such an events import. Later another change added a type (not implementation) dependency on node type. This caused the actual type dependency for the "events" import to be on node version (which is slightly different) but was uncaught (as the prior false positive was suppressed). This project did not intend to have any node dependencies.

**Comment by fisker** [comment]
Mabay you can import as `events/`?  https://github.com/sindresorhus/eslint-plugin-unicorn/issues/1931
